### ISSUE TITLE
segfault from particular indexing of empty array (migrated from Trac #607)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#1214]
Original ticket http://projects.scipy.org/numpy/ticket/607
Reported 2007-11-06 by atmention:efiring, assigned to unknown.

The following causes a segfault upon execution as a script:

import numpy
u = numpy.array([])
g = numpy.array([True, True])
u[g] = 0

See the thread starting with
http://projects.scipy.org/pipermail/numpy-discussion/2007-November/029801.html

**Comment by thouis** [comment]
Attachment in Trac by atmention:efiring, 2007-11-06: [numpysegfault.py](http://projects.scipy.org/numpy/attachment/ticket/607/numpysegfault.py)

**Comment by thouis** [comment]
Comment in Trac by atmention:efiring, 2007-11-06

Try again with formatting:

```
import numpy
u = numpy.array([])
g = numpy.array([True, True])
u[g] = 0
```

**Comment by thouis** [comment]
Comment in Trac by atmention:nilswagner01, 2007-11-06

I cannot reproduce the segfault with 

```
>>> numpy.__version__
'1.0.5.dev4399'
```

**Comment by thouis** [comment]
Comment in Trac by atmention:efiring, 2007-11-18

This is fixed as of 1.0.5.dev4466.

**Comment by thouis** [comment]
Comment in Trac by atmention:stefanv, 2007-11-18

Fixed in r4466.
